lib/xcodeproj/project/object/file_system_synchronized_root_group.rb

```diff
@@ -0,0 +1,15 @@
+module Xcodeproj
+  class Project
+    module Object
+      # This class represents a file system synchronized root group.
+      class PBXFileSystemSynchronizedRootGroup < AbstractObject
```

Comment:
@Brett-Best thank you for working on this! 

Unfortunately, from what I was able to gather, the implementation is a tiny bit incomplete. 

1. There are a few attributes that `PBXFileSystemSynchronizedRootGroup` has in common with `PBXGroup`, such as tabs/indent properties. 

2. There are also some properties unique to `PBXFileSystemSynchronizedRootGroup` that need to be added, such as `explicitFileTypes`, `explicitFolders` and exceptions (which require a further `PBXFileSystemSynchronizedBuildFileExceptionSet` object to represent).

Without this, I believe the representation will be incomplete and cause crashes.

I can recommend [this article on synchronized groups](https://pepicrft.me/blog/2024/07/20/how-synchronized-groups-work-at-the-pbxproj-level) by @pepicrft or [this Tuist PR](https://github.com/tuist/XcodeProj/pull/827) by the same for further info.

Reply:
Thanks for the reply, yeah, it would be nice if the project file specification was published somewhere 🙃.
Yep, I expect if the project file uses these additional attributes that warnings/errors/crashes should be emitted.

I basically just added support to unblock the current utilization of the `PBXFileSystemSynchronizedRootGroup` in a project I was working on.

I may or may not end up updating this PR with the additional types, given CocoaPods is now in maintenance mode 😄.

Unfortunately, can't migrate off of CocoaPods to SPM yet because React Native 😢

Reply:
Thanks for the reply, I've run into a new issue:

When a build configuration uses an xcconfig in the new folders the baseConfigurationReference is removed and replaced with baseConfigurationReferenceAnchor (PBXFileSystemSynchronizedRootGroup) and baseConfigurationReferenceRelativePath.

I managed to add that to the `xcodeproj` mostly fine but on the cocoapods side it needs some love at least in this section:
https://github.com/CocoaPods/CocoaPods/blob/85e6d43ca63f8acf0bb3f3d910e8ff1f6558a4e8/lib/cocoapods/installer/user_project_integrator/target_integrator/xcconfig_integrator.rb#L44-L66

Reply:
Another issue is that my change adds `file_system_synchronized_groups` to all entries which is fine but when you edit the project file in Xcode, it removes the entries that are empty arrays.

Would you have any idea how I can stop Xcodeproj from adding empty array entries?

Reply:
Same issue above with the `exceptions` key of a `PBXFileSystemSynchronizedRootGroup`.

Reply:
@Brett-Best I've looked at the rest of the uses of `has_many` - I think it's not possible to avoid initializing the empty `ObjectList`. Is this causing any issues? From what I know, many other things generated by `Xcodeproj` get modified by Xcode, and this has never been a concern.

In any case, great work! There are just a few more attributes to add, I think all of those are simple value types:

* Copy attributes from `PBXGroup` to `PBXFileSystemSynchronizedRootGroup`: `name`, `uses_tabs`, `indent_width`, `tab_width`, `wraps_lines`
* Add new attributes to `PBXFileSystemSynchronizedRootGroup`: 
  * `explicit_file_types` (type `Hash`)
  * `explicit_folders` (type `Array`)
* Add new attributes to `PBXFileSystemSynchronizedBuildFileExceptionSet`:
  * `public_headers` (type `Array`)
  * `private_headers` (type `Array`)
  * `additional_compiler_flags_by_relative_path` (type `Hash`)
  * `attributes_by_relative_path` (type `Hash`)

Reply:
@igor-makarov along with the additional attributes listed, hopefully they don't also add empty groups.
Yes it is an issue, other things in the `Pod` project can sometimes change if you open it in Xcode to inspect the settings etc but this far less likely to be done on a day to day basis.
Whereas changing the settings of your own project is much more common.
Where I work atm we have churn of this getting added in / removed.
As a workaround I did come up with some regex but I couldn't get it to work on shell cause multiline hah...